### ISSUE TITLE
Unspecified Certificate Verification Error

### ISSUE THREAD

**Issue by Grime121** [looker-open-source/gzr#104]
Even when using the —no-verify-ssl option, we are still getting a certificate verification error:

/usr/lib/ruby/2.7.0/net/protocol.rb:44:in `connect_nonblock': SSL_connect returned=1 errno=0 state=error: certificate verify failed (unspecified certificate verification error) (Faraday::SSLError)

I have verified that both the root and intermediate CA certificates are installed. The problem with the certificate is it is for our public URL, but we use the internal URL for Gazer, as we do not want to expose it to the Internet. So, the name on the certificate does not match the URL.

Shouldn’t using the —no-verify-ssl option prevent us from getting certificate verification errors? If so, is this a bug…? It not, what do I need to do to fix this? Is there a way to specify a different certificate for the Gazer API website?

**Comment by Grime121** [comment]
Does anyone have any idea as to why —no-verify-ssl isn’t preventing certificate errors from causing Gazer connections to fail…? I’m almost certain the —no-verify-ssl option is simply not working, now that I have exposed the Gazer API to the Internet, using our public URL. Once I did that I was able to successfully connect to it from a client that had the root and intermediate CA certificates installed. I still cannot connect to it from a client that does not have the intermediate CA installed, though. So, obviously the —no-verify-ssl option is failing to allow connections to Gazer APIs that the client does not have the CA certs for, AND it is not allowing connections from clients that have the CA certs installed, but the client is using a hostname other than the one on the certificate to connect to the Gazer API. If the —no-verify-ssl option does not help with either of those situations, what IS it intended to help with?
